### PR TITLE
feat(worker): Install and use ssh key in a step script #3022

### LINKED ISSUE #3022: Install and use ssh key in a step script

### PR DESCRIPTION
**Pull request by bnjjj**
1. Description
1. Related issues
close #3022 
1. About tests
1. Mentions
cc @jlejeune 

@ovh/cds

### CODE REVIEW

File: engine/worker/cmd_key.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			Message: fmt.Sprintf("Cannot setup ssh key %s : %v", keyName, err),
+			Status:  http.StatusInternalServerError,
+		}
+		log.Error("%v", errSetup)
+		writeJSON(w, errSetup, errSetup.Status)
+		return
+	}
+
+	writeJSON(w, keyResponse{PKey: wk.currentJob.pkey}, http.StatusOK)
+	return
```
Comment (golangci) [inline engine/worker/cmd_key.go]: redundant return statement

Comment (golangci) [inline engine/worker/cmd_key.go]: redundant return statement

File: docs/content/workflows/pipelines/actions/builtin/script.md [hunk 2]
```diff
@@ -63,6 +63,12 @@ You can use [CDS Variables]({{< relref "workflows/pipelines/variables.md" >}}) i
 
 ![img](/images/workflows.pipelines.actions.builtin.script-bash.png)
 
+#### Using CDS keys in a script
+
+You can use CDS SSH keys in a step script with the [worker key command]({{< relref "cli/worker/key/_index.md" >}}). Just use `worker key install proj-mykey` and it will install the SSH environment and private SSH key of your key in your project named **mykey**.
+
+The command `worker key install proj-mykey` will return the path where the private key is stored. In that way you can save this value in a variable and use it for a ssh command like this `ssh -i $MY_VAR_WITH_PATH_TO_MY_PRIVATE_KEY`
+
```
Comment (fsamin) [inline docs/content/workflows/pipelines/actions/builtin/script.md]: why not display the whole bash example ?

File: engine/worker/cmd_key.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		req, errRequest := http.NewRequest(
+			"POST",
+			fmt.Sprintf("http://127.0.0.1:%d/key/%s/install", port, url.PathEscape(args[0])),
+			bytes.NewReader(nil),
+		)
+		if errRequest != nil {
+			sdk.Exit("Error: worker key install > cannot post worker key install (Request): %s\n", errRequest)
+		}
+
+		resp, errDo := http.DefaultClient.Do(req)
```
Comment (fsamin) [inline engine/worker/cmd_key.go]: no timeout ?

File: engine/worker/cmd_key.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode >= 300 {
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				sdk.Exit("Error: worker key install> HTTP error %v\n", err)
+			}
+			cdsError := sdk.DecodeError(body)
+			sdk.Exit("Error: worker key install> http code %d : %v\n", resp.StatusCode, cdsError)
```
Comment (fsamin) [inline engine/worker/cmd_key.go]: i think we don't need the resp.StatusCode

**Comment by ovh-cds** [comment]
CDS Report it#5062.2 ✘  
* w-it
  * w-it ✘  
Unit Tests Report
* Platform model admin [tests/cli_admin_platform_model.yml]




* Application Command TestSuite [tests/cli_application.yml]









* Environment Command TestSuite [tests/cli_environment.yml]














* Pipeline Command TestSuite [tests/cli_pipeline.yml]
















* Pipeline Export / Import Command TestSuite [tests/cli_pipeline_export_import.yml]














* Action Command TestSuite with cdsctl [tests/clictl_action.yml]







* Broadcast Command TestSuite [tests/clictl_admin_broadcast.yml]


* Plugins Command TestSuite with cdsctl [tests/clictl_admin_plugin_and_platforms.yml]










* Action Command TestSuite with cdsctl [tests/clictl_admin_services.yml]


* Application Command TestSuite with CDS ctl [tests/clictl_application.yml]









* Environement Command TestSuite with CDS ctl [tests/clictl_environment.yml]









* Group Command TestSuite [tests/clictl_group.yml]












* Pipeline Command TestSuite with CDS ctl [tests/clictl_pipeline.yml]









* Project Command TestSuite with CDS ctl [tests/clictl_project.yml]












* Token Command TestSuite with CDS ctl [tests/clictl_token.yml]







* User Command TestSuite [tests/clictl_user.yml]


* Version  Command TestSuite [tests/clictl_version.yml]


* Worker Command TestSuite [tests/clictl_worker_model.yml]













* Create a simple workflow (ITSCWRKFLW3) and run it [tests/sc_workflow_create_run_join.yml]



* Create a simple workflow (ITSCWRKFLW1) and pull it [tests/sc_workflow_create_simple.yml]






* Workflow (ITSCWRKFLW5) should use the worker cache command [tests/sc_workflow_key_install.yml]




  * the workflow should be successfull ✘

* Workflow (ITSCWRKFLW4) should use the worker cache command [tests/sc_workflow_run_cache.yml]






* Create a simple workflow (ITSCWRKFLW2) and run it [tests/sc_workflow_run_simple.yml]







* Create a simple workflow (ITSCWRKFLW2) run it and then stop it [tests/sc_workflow_stop_simple.yml]








* Create a simple workflow (ITSCWRKFLW2), update it and run it [tests/sc_workflow_update_run.yml]

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5067.0 ✘  
* Stage 1
  * Engine Test with DB ✘  
Unit Tests Report
* github.com/ovh/cds/cli


* github.com/ovh/cds/engine/api





































































































































































































  * Test_postWorkflowHandlerWithRootShouldSuccess ✘

* github.com/ovh/cds/engine/api/application








* github.com/ovh/cds/engine/api/environment




* github.com/ovh/cds/engine/api/keys



* github.com/ovh/cds/engine/api/migrate


* github.com/ovh/cds/engine/api/pipeline






* github.com/ovh/cds/engine/api/plugin


* github.com/ovh/cds/engine/api/poller




* github.com/ovh/cds/engine/api/project









* github.com/ovh/cds/engine/api/scheduler












* github.com/ovh/cds/engine/api/secret






* github.com/ovh/cds/engine/api/user








* github.com/ovh/cds/engine/api/warning















* github.com/ovh/cds/engine/api/worker










* github.com/ovh/cds/engine/api/workflow












































* github.com/ovh/cds/engine/hatchery/swarm



















* github.com/ovh/cds/engine/hooks






* github.com/ovh/cds/engine/repositories


* github.com/ovh/cds/engine/vcs











* github.com/ovh/cds/engine/vcs/bitbucket


















* github.com/ovh/cds/engine/vcs/github













* github.com/ovh/cds/engine/vcs/gitlab









* github.com/ovh/cds/engine/worker






* github.com/ovh/cds/sdk















* github.com/ovh/cds/sdk/exportentities




























* github.com/ovh/cds/sdk/interpolate

























* github.com/ovh/cds/sdk/luascript




* github.com/ovh/cds/sdk/plugin/dummy/caller


* github.com/ovh/cds/sdk/vcs/git

**Pull request merged**